lib/gitsh/tab_completion/matchers/path_matcher.rb

```diff
@@ -0,0 +1,34 @@
+require 'gitsh/tab_completion/matchers/base_matcher'
+
+module Gitsh
+  module TabCompletion
+    module Matchers
+      class PathMatcher < BaseMatcher
+        def completions(token)
+          prefix = normalize_path(token)
+          paths(prefix).map { |option| option.sub(prefix, token) }
+        end
+
+        private
+
+        def normalize_path(token)
+          if token.end_with?('/')
+            File.expand_path(token) + '/'
+          else
+            File.expand_path(token)
+          end
+        end
+
+        def paths(prefix)
+          Dir["#{prefix}*"].map do |path|
+            if File.directory?(path)
+              path + '/'
```

Comment:
I think it is a bit too much: if I saw this in practice, I'd need to read the `conditionally_slash` method to understand what was going on, so I'm going to leave it inline.